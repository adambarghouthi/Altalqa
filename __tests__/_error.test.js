import React from 'react';
import ErrorPage from '../src/pages/_error';
import renderer from 'react-test-renderer';

describe('ErrorPage should show', () => {
	test('server error if statusCode exists', () => {
		const component = renderer.create(<ErrorPage statusCode={500} />)
		let tree = component.toTree()

		expect(tree).toMatchSnapshot()
		expect(tree.rendered.type).toBe('p')
		expect(tree.props.statusCode).toEqual(500)
		expect(tree.rendered.props.children).toBe('An error 500 occurred on server')
	})

	test('client error if statusCode undefined', () => {
		const component = renderer.create(<ErrorPage />)
		let tree = component.toTree()

		expect(tree).toMatchSnapshot()
		expect(tree.rendered.type).toBe('p')
		expect(tree.props.statusCode).toEqual(undefined)
		expect(tree.rendered.props.children).toBe('An error occurred on client')
	})
})
