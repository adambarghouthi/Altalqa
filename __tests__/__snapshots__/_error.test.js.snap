// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should show client error if statusCode undefined 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": "An error occurred on client",
    },
    "rendered": Array [
      "An error occurred on client",
    ],
    "type": "p",
  },
  "type": [Function],
}
`;

exports[`ErrorPage should show server error if statusCode exists 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "statusCode": 500,
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": "An error 500 occurred on server",
    },
    "rendered": Array [
      "An error 500 occurred on server",
    ],
    "type": "p",
  },
  "type": [Function],
}
`;
